addons/web/controllers/domain.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http, _
+from odoo.http import Controller, request
+from odoo.exceptions import ValidationError
+from odoo.tools.misc import mute_logger
+
+class Domain(Controller):
+
+    @http.route('/web/domain/validate', type='json', auth="user")
+    def validate(self, model, domain):
+        """ Parse `domain` and verify that it can be used to search on `model`
+        :return: True when the domain is valid, otherwise False
+        A validation error is raised if `model` is invalid
+        """
+        try:
+            Model = request.env[model]
+        except Exception:
+            raise ValidationError(_('Invalid model: %s', model))
+        try:
+            # go through the motions of preparing the final SQL for the domain,
+            # so that anything invalid will raise an exception.
+            query = Model._search(domain)
```

Comment:
You don't need to test access rules, so leave them out.
```suggestion
            query = Model.sudo()._search(domain)
```